workers/unity/Packages/io.improbable.gdk.debug/WorkerInspector/EntityList.cs

```diff
@@ -0,0 +1,268 @@
+            private World world;
+            private EntityQuery query;
+
+            private readonly EntityQueryDesc queryDesc = new EntityQueryDesc
+            {
+                All = new[] { ComponentType.ReadOnly<SpatialEntityId>() }
+            };
+
+            public void Calculate()
+            {
+                using (applyFilterMarker.Auto())
+                {
+                    Data.Clear();
+
+                    foreach (var datum in underlying)
+                    {
+                        if (datum.Matches(SearchParameters))
+                        {
+                            Data.Add(datum);
+                        }
+                    }
+                }
+            }
+
+            public void RefreshData(World newWorld)
+            {
+                if (newWorld == null)
+                {
+                    world = null;
+                    underlying = new List<EntityData>();
+                    return;
+                }
+
+                if (world != newWorld)
+                {
+                    // Need to refresh the query.
+                    world = newWorld;
+                    query = world.EntityManager.CreateEntityQuery(queryDesc);
+                }
+
+                using (refreshDataMarker.Auto())
+                {
+                    underlying.Clear();
+                    var spatialOSComponentType = world.EntityManager.GetArchetypeChunkComponentType<SpatialEntityId>(true);
+                    var metadataComponentType = world.EntityManager.GetArchetypeChunkComponentType<Metadata.Component>(true);
+                    var ecsEntityType = world.EntityManager.GetArchetypeChunkEntityType();
+
+                    using (var chunks = query.CreateArchetypeChunkArray(Allocator.TempJob))
+                    {
+                        foreach (var chunk in chunks)
+                        {
+                            NativeArray<Metadata.Component>? metadataArray = null;
+
+                            if (chunk.Has(metadataComponentType))
+                            {
+                                metadataArray = chunk.GetNativeArray(metadataComponentType);
+                            }
+
+                            var entityIdArray = chunk.GetNativeArray(spatialOSComponentType);
+                            var entities = chunk.GetNativeArray(ecsEntityType);
+
+                            for (var i = 0; i < entities.Length; i++)
+                            {
+                                var data = new EntityData(entityIdArray[i].EntityId, metadataArray?[i].EntityType);
+                                underlying.Add(data);
+                            }
+                        }
+                    }
+                }
+
+                underlying.Sort();
+
+                Calculate();
+            }
+        }
+
+        private readonly struct EntityData : IComparable<EntityData>, IComparable, IEquatable<EntityData>
```

Comment:
`IComparable` seems extra as this is implemented by `IComparable<T>` ?

Reply:
This was the Rider auto-generated impl